### PR TITLE
Improve error messaging for ASR pipeline.

### PR DESCRIPTION
**Pull request by Narsil**
# What does this PR do?

- ~~Raise error early (in `_sanitize`) so users don't waste time trying to
  run queries with invalid params.~~ This is not easy unfortunately because the order of resolution of objection is tricky.

- Fix the error was after using `config.inputs_to_logits_ratio` so our
  check was masked by the failing property does not exist.

- Added some manual check on s2t for the error message.
  No non ctc model seems to be used by the default runner (they are all
  skipped).

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

@sgugger  @ArthurZucker 
Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

File: tests/pipelines/test_pipelines_automatic_speech_recognition.py [hunk 1]
```diff
@@ -138,6 +144,17 @@ def test_small_model_pt(self):
...
+            _ = speech_recognizer(waveform, chunk_length_s=10)
+        self.assertEqual(
+            str(v.exception),
+            "`chunk_length_s` is only valid for CTC models, use other chunking options for other models",
+        )
+
+        # Non CTC models cannot use return_timestamps
+        with self.assertRaises(ValueError) as v:
+            _ = speech_recognizer(waveform, return_timestamps="char")
+        self.assertEqual(str(v.exception), "We cannot return_timestamps yet on non-ctc models !")
```
Comment (ArthurZucker) [inline tests/pipelines/test_pipelines_automatic_speech_recognition.py:157]: NIce test 😉

**Review by ArthurZucker — APPROVED** [review]
Good catch, will close #19490 !

**Review by sgugger — APPROVED** [review]
Thanks for improving this error message!
